p2p/src/peer_manager/mod.rs

```diff
@@ -313,26 +350,37 @@ where
         // accepted even if it's valid. The peer is still reported to the PeerDb which
         // knows of all peers and later on if the number of connections falls below
         // the desired threshold, `PeerManager::heartbeat()` may connect to this peer.
-        if self.peerdb.active_peer_count() >= MAX_ACTIVE_CONNECTIONS {
+        if self.active_peer_count() >= MAX_ACTIVE_CONNECTIONS {
             return Err(P2pError::PeerError(PeerError::TooManyPeers));
         }
 
         self.accept_connection(address, Role::Inbound, info, receiver_address).await
     }
 
-    /// Close connection to a remote peer
+    /// Connection to a remote peer closed
     ///
     /// The decision to close the connection is made either by the user via RPC
```

Comment:
This is kind of a heavy promise to make in a comment... I'm confused. This is a state-machine hell.

Reply:
Sorry, my bad, I forgot to update the comment here.
I think the connect/disconnect flow was broken before.
`PeerManager` will now update itself only from events from the backend.
Connect and disconnect flows are consistent and simple now:
got connect/disconnect request => sent request to backend => received event from backend => state updated.
For example, this should work correctly if `PeerManager` sends a disconnect request but the peer drops own connection before backend have a chance to process the request.